File: frontend/packages/common/chat-area/chat-uikit-shared/src/types/event.ts
/*
 * Copyright 2025 coze-dev Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { type MouseEvent } from 'react';

import { type SendMessageOptions } from '@coze-common/chat-core';

import { type IMessage } from './common';

export interface IEventCallbacksParams<T = Record<string, unknown>> {
  message: IMessage;
  extra: T;
}

export interface LinkEventData {
  url: string;
  parsedUrl: URL;
  exts: {
    // type: LinkType;
    wiki_link?: string;
  };
}

export type IOnLinkClickParams = IEventCallbacksParams<LinkEventData>;

export type IOnImageClickParams = IEventCallbacksParams<{
  url: string;
}>;

export type IOnCancelUploadParams = IEventCallbacksParams;

export type IOnRetryUploadParams = IEventCallbacksParams;

export type IOnSuggestionClickParams = IEventCallbacksParams<{
  text: string;
  mentionList: { id: string }[];
}>;

export type IOnMessageRetryParams = IEventCallbacksParams;

export type IOnCopyUploadParams = IEventCallbacksParams<{ fileIndex?: number }>;

export type IOnCardSendMsg = IEventCallbacksParams<{
  msg: string;
  mentionList: { id: string }[];
  options?: SendMessageOptions;
}>;
export type IOnCardUpdateStatus = IEventCallbacksParams;

export interface MouseEventProps {
  element: HTMLElement;
  link: string;
}

export type IEventCallbacks = Partial<{
  // Click the callback function linked in md
  onLinkClick: (
    params: IOnLinkClickParams,
    event: MouseEvent<Element, globalThis.MouseEvent>,
  ) => void;
  // image click incident response
  onImageClick: (params: IOnImageClickParams) => void;
  // Cancel upload event response
  onCancelUpload: (params: IOnCancelUploadParams) => void;
  // Resume upload incident response
  onRetryUpload: (params: IOnRetryUploadParams) => void;
  // Copy file link
  onCopyUpload: (params: IOnCopyUploadParams) => void;
  // resend
  onMessageRetry: (params: IOnMessageRetryParams) => void;
  // Click the card button for incident response
  onCardSendMsg: (params: IOnCardSendMsg) => void;
  // Update card status
  onCardUpdateStatus: (params: IOnCardUpdateStatus) => void;
  onCardLinkElementEnter?: (params: MouseEventProps) => void;
  onCardLinkElementLeave?: (params: MouseEventProps) => void;
  onMdBoxLinkElementEnter?: (params: MouseEventProps) => void;
  onMdBoxLinkElementLeave?: (params: MouseEventProps) => void;
  onMdBoxImageElementEnter?: (params: MouseEventProps) => void;
  onMdBoxImageElementLeave?: (params: MouseEventProps) => void;
}>;

File: backend/domain/knowledge/entity/event.go
/*
 * Copyright 2025 coze-dev Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package entity

type Event struct {
	Type EventType

	Documents      []*Document
	Document       *Document
	Slice          *Slice
	SliceIDs       []int64
	KnowledgeID    int64
	DocumentReview *Review
}

type EventType string

// Document event
// Split + write vector library operation transactionality is guaranteed by the implementation itself
const (
	EventTypeIndexDocuments EventType = "index_documents"

	// EventTypeIndexDocument document information has been written to orm, the logic needs to parse + split + search data warehousing
	// Event requires: Event.Document
	EventTypeIndexDocument EventType = "index_document"

	// EventTypeIndexSlice slice information has been written to orm, and only search data is written in the logic
	// Event requires: Event.Slice
	EventTypeIndexSlice EventType = "index_slice"

	// EventTypeDeleteKnowledgeData remove knowledge
	// Event requires: Event.KnowledgeID, Event.SliceIDs
	EventTypeDeleteKnowledgeData EventType = "delete_knowledge_data"

	EventTypeDocumentReview EventType = "document_review"
)
